app/controllers/candidate_interface/course_choices_controller.rb

```diff
@@ -146,9 +146,14 @@ def destroy
     def complete
       @application_form = current_application
 
-      if @application_form.application_choices.count.zero?
-        render :index
-      elsif @application_form.update(application_form_params)
+      render :index if @application_form.application_choices.count.zero?
+
+      if @application_form.submitted?
+        edit_by_time = TimeLimitCalculator.new(rule: :edit_by, effective_date: @application_form.submitted_at).call[:time_in_future]
+        @application_form.application_choices.each { |choice| choice.update!(edit_by: edit_by_time) }
```

Comment:
Maybe we should call the `SubmitApplication` service here? I just noticed the state is still unsubmitted, not sure if it would cause other issues. Any suggestions?

Reply:
This is an interesting one. I wonder if we should refactor `SubmitApplication` to call a `SubmitApplicationChoice` service, and call that service from here too?

Reply:
Yes, I had a similar idea in my head this morning! I'll have a look

Reply:
I made some changes, IMO it helped, what do you think? @duncanjbrown